### PR TITLE
feat(destination): singular- add support for other apple os

### PR DESCRIPTION
**Pull request by yashasvibajpai**
## Description of the change

> Allow sending ios, ipados, watchos or tvos in contextual properties 

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by krishna2020 — APPROVED** [review]

File: v0/destinations/singular/util.js [hunk 1]
```diff
@@ -19,7 +19,8 @@ const {
   extractCustomFields,
   getValueFromMessage,
   isDefinedAndNotNull,
-  toUnixTimestamp
+  toUnixTimestamp,
```
Comment (ItsSudip) [inline v0/destinations/singular/util.js:22]: We are not using `toUnixTimestamp` so we can remove this.

File: v0/destinations/singular/util.js [hunk 2]
```diff
@@ -109,6 +110,10 @@ const platformWisePayloadGenerator = (message, isSessionEvent) => {
     throw new CustomError("[Singular] :: Platform name is missing", 400);
   }
   platform = platform.toLowerCase();
+  // checking if the os is one of ios, ipados, watchos, tvos
+  if (isAppleFamily(platform)) {
+    message.context.os.name = "iOS";
```
Comment (ItsSudip) [inline v0/destinations/singular/util.js:115]: We can directly update the platform to "IOS".

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: v0/destinations/singular/util.js (hunk 2 above)
Comment (yashasvibajpai) [reply v0/destinations/singular/util.js:115]: but platform is not being used to set in the mapping right, rather is `context.os.name`

Comment (ItsSudip) [reply v0/destinations/singular/util.js:115]: Yes but along with that we can update platform as well. No need to update supported_platform in /config.js, as we are doing the same mapping as `isAppleFamily`.

Comment (yashasvibajpai) [reply v0/destinations/singular/util.js:113]: addressed

File: v0/destinations/singular/util.js (hunk 1 above)
Comment (yashasvibajpai) [reply v0/destinations/singular/util.js]: addressed

File: v0/destinations/singular/util.js [hunk 3]
```diff
@@ -108,6 +108,11 @@ const platformWisePayloadGenerator = (message, isSessionEvent) => {
   if (!platform) {
     throw new CustomError("[Singular] :: Platform name is missing", 400);
   }
+  // checking if the os is one of ios, ipados, watchos, tvos
+  if (isAppleFamily(platform)) {
```
Comment (ItsSudip) [inline v0/destinations/singular/util.js:112]:
```suggestion
  if (typeof platform === 'string' && isAppleFamily(platform.toLowerCase())) {
```

**Review by ItsSudip — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by krishna2020 — APPROVED** [review]

**Pull request merged**